error C2124: divide or mod by zero

## Bug report

**Describe the bug**

I tried to use libvips with MSVC but I got this compile error.
```
../subprojects/libvips/libvips/arithmetic/complex.c(490): error C2124: divide or mod by zero
```

To investigate this error I tried expanding the macro in line 490 manually like the following, and the error indeed seems to occur at `double a = 0.0 / 0.0;`.

```
static void
vips_complex2_buffer(VipsArithmetic *arithmetic,
    VipsPel *out, VipsPel **in, int width)
{
    VipsComplex2 *cmplx = (VipsComplex2 *) arithmetic;
    VipsImage *im = arithmetic->ready[0];
    const int sz = width * vips_image_get_bands(im);

    int x;

    switch (cmplx->cmplx) {
    case VIPS_OPERATION_COMPLEX2_CROSS_PHASE:
        //SWITCH2(CROSS);
        switch (vips_image_get_format(im)) {
        case VIPS_FORMAT_UCHAR:
            //LOOP2(unsigned char, float, CROSS); -> (IN, OUT, OP)
            {
                typedef unsigned char IN;
                typedef float OUT;

                IN *p1 = (IN *) in[0];
                IN *p2 = (IN *) in[1];
                OUT *q = (OUT *) out;

                for (x = 0; x < sz; x++) {
                    //CROSS(q, p1[x], 0.0, p2[x], 0.0); -> (Q, X1, Y1, X2, Y2)
                    {
                        if (((p1[x]) == 0.0 && (0.0) == 0.0) ||
                            ((p2[x]) == 0.0 && (0.0) == 0.0)) {
                            q[0] = 0.0;
                            q[1] = 0.0;
                        }
                        else if (ABS(0.0) > ABS(0.0)) {
                            double a = 0.0 / 0.0;
                            double b = 0.0 + 0.0 * a;
                            double re = (p1[x] + p2[x] * a) / b;
                            double im = (p2[x] - p1[x] * a) / b;
                            double mod = hypot(re, im);

                            q[0] = re / mod;
                            q[1] = im / mod;
                        }
                        else {
                            double a = 0.0 / 0.0;
                            double b = 0.0 + 0.0 * a;
                            double re = (p1[x] * a + p2[x]) / b;
                            double im = (p2[x] * a - p1[x]) / b;
                            double mod = hypot(re, im);

                            q[0] = re / mod;
                            q[1] = im / mod;
                        }
                    }


                    q += 2;
                }
            }
            break;
```

Is this code correct?
I have no idea what this calculation doing, but I could workaround this error by putting X1, Y1, X2, Y2 value into a variable like this https://github.com/na-trium-144/libvips/commit/33023cab219d394604afa078723d70b0714f324c.
I can submit this as PR if this workaround is right.

**To Reproduce**
Steps to reproduce the behavior:
1. Try to build libvips using Meson with Visual Studio 2022 or 2019.

**Expected behavior**
libvips compiles without error.

**Actual behavior**
Cannot compile due to error C2124.

**Screenshots**

**Environment**
(please complete the following information)
 - OS: Windows 11
 - Vips: current master branch https://github.com/libvips/libvips/commit/0c9c7b16fe8ed35237c8994c13ead654f94ed860

**Additional context**

--- a/libvips/arithmetic/complex.c
+++ b/libvips/arithmetic/complex.c
@@ -455,7 +455,8 @@ G_DEFINE_TYPE(VipsComplex2, vips_complex2, VIPS_TYPE_BINARY);
 			Q[1] = 0.0; \
 		} \
 		else if (ABS(Y1) > ABS(Y2)) { \
-			double a = Y2 / Y1; \
+			double y1 = Y1; /* this suppress C2142 (division by zero) error on MSVC */ \
+			double a = Y2 / y1; \
 			double b = Y1 + Y2 * a; \
 			double re = (X1 + X2 * a) / b; \
 			double im = (X2 - X1 * a) / b; \
@@ -465,7 +466,8 @@ G_DEFINE_TYPE(VipsComplex2, vips_complex2, VIPS_TYPE_BINARY);
 			Q[1] = im / mod; \
 		} \
 		else { \
-			double a = Y1 / Y2; \
+			double y2 = Y2; \
+			double a = Y1 / y2; \
 			double b = Y2 + Y1 * a; \
 			double re = (X1 * a + X2) / b; \
 			double im = (X2 * a - X1) / b; \

--- a/libvips/arithmetic/complex.c
+++ b/libvips/arithmetic/complex.c
@@ -449,7 +449,8 @@ G_DEFINE_TYPE(VipsComplex2, vips_complex2, VIPS_TYPE_BINARY);
 #define CROSS(Q, X1, Y1, X2, Y2) \
 	{ \
 		if (((X1) == 0.0 && (Y1) == 0.0) || \
-			((X2) == 0.0 && (Y2) == 0.0)) { \
+			((X2) == 0.0 && (Y2) == 0.0) || \
+			((Y1) == 0.0 && (Y2) == 0.0)) { \
 			Q[0] = 0.0; \
 			Q[1] = 0.0; \
 		} \

